data.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/lib/pq"
+)
+
+// lookup table for buildings that we have in our system.
+var parentNameLookup = map[int]string{
+	146: "Avery",
+	103: "Butler",
+	62:  "East Asian Library",
+	75:  "John Jay",
+	79:  "Lehman Library",
+	84:  "Lerner",
+	15:  "Northwest Corner Building",
+	2:   "Uris",
+}
+
+// dataset is an alias for an array of data dumps
+type dataset []dumpFormat
+
+// dumpFormat represents the datapoints provided for the wireless data.
+// Some fields are gathered from the dumped JSON files and others are configured
+// afterwards before storage.
+type dumpFormat struct {
+	DumpTime    time.Time
+	GroupID     int
+	ParentName  string
+	GroupName   string `json:"name"`
+	ParentID    int    `json:"parent_id"`
+	ClientCount int    `json:"client_count"`
+}
+
+// parseData unmarshals a byte array into an array of wireless data dumps.
+//
+// This is a little more complicated because the group ID is stored as the key to the
+// remainder of the data for the record.
+//
+// adds:
+// - a timestamp based on the filename
+// - a group ID based on the group's key in the JSON
+// - a parent name based on the parentNameLookup table
+func parseData(timestamp time.Time, datafile []byte) (dataset, error) {
+	// marshal what data we can from the json
+	parsed := make(map[string]dumpFormat)
+	if err := json.Unmarshal(datafile, &parsed); err != nil {
+		return []dumpFormat{}, fmt.Errorf("Error parsing bytes => %s", err.Error())
+	}
+
+	var (
+		data   = make([]dumpFormat, len(parsed))
+		i      int
+		err    error
+		exists bool
+	)
+	// fill out the JSON with the group ID added
+	for id, d := range parsed {
+		if d.GroupID, err = strconv.Atoi(id); err != nil {
```

Comment:
In these combined statement, is only the `err != nil` statement evaluated for truthiness?